sources/kube_factory.go

```diff
@@ -94,10 +95,28 @@ func CreateKubeSources(uri string, options map[string][]string) ([]api.Source, e
 			return nil, err
 		}
 	}
+	kubeletHttps := defaultKubeletHttps
+	if len(options["kubeletHttps"]) >= 1 {
+		kubeletHttps, err = strconv.ParseBool(options["kubeletHttps"][0])
+		if err != nil {
+			return nil, err
+		}
+	}
 	glog.Infof("Using Kubernetes client with master %q and version %s\n", kubeConfig.Host, kubeConfig.Version)
 	glog.Infof("Using kubelet port %d", kubeletPort)
-	kubeletApi := datasource.NewKubelet()
-	kubePodsSource := NewKubePodMetrics(kubeletPort, nodesApi, newPodsApi(kubeClient), kubeletApi)
+
+	kubeletConfig := &kube_client.KubeletConfig{
+		Port:            uint(kubeletPort),
+		EnableHttps:     kubeletHttps,
+		TLSClientConfig: kubeConfig.TLSClientConfig,
```

Comment:
No, I wasn't even aware of #269. 

I'm runnin heapster like this: heapster --source="kubernetes:https://ip:port?auth=clientauth&kubeletPort=10250", where clientauth is a json file that defines paths to the cert files (docs on this are here: https://github.com/GoogleCloudPlatform/heapster/blob/master/docs/source-configuration.md)